cvmfs/download.cc

```diff
@@ -1879,14 +1902,38 @@ bool DownloadManager::ProbeHostsGeo() {
   }
   if (!success) {
     LogCvmfs(kLogDownload, kLogDebug | kLogSyslogWarn,
-             "failed to retrieve geographic order of stratum 1 servers");
+             "failed to retrieve geographic order from stratum 1 servers");
     return false;
   }
 
   pthread_mutex_lock(lock_options_);
   delete opt_host_chain_;
+  opt_host_chain_ = new vector<string>(host_chain.size());
+
+  vector< vector< ProxyInfo> > *proxy_groups = 
+        new vector< vector<ProxyInfo> > (
+            opt_proxy_groups_fallback_ + proxy_chain.size() - fallback_group);
```

Comment:
Ah, I see.  That means that `opt_proxy_fallback_list_`, `opt_num_proxies_`, `opt_proxy_groups_current_burned_`, and `opt_proxy_groups_current_` are possibly out of sync, too.  But should be possible to reset them here.